### PR TITLE
feat: add wind_field endpoint and corresponding tests

### PR DESCRIPTION
**Pull request by bilal-sattar-nats**
- Added a new wind endpoint in the API
- Added API coverage for that endpoint
- Refactored WindField serialization/deserialization from custom construct-data JSON to full model-based JSON
- WindField now carries optional wind_speed and wind_direction fields directly, with validators/serializers, and uses those in equality and fl_interpolation logic
- Expanded wind unit tests around JSON round-trips, stability, and validation expectations
- Remove obsolete observe() function in the environment manager, since sector_id was unused. As a result, removed environment endpoint that depended on sector id and depending tests,

Closes project-bluebird/BluebirdATC#62 and #68

### CODE REVIEW

File: bluebird-api/bluebird_api/routers/core.py
Comment (AlvaroSierra) [inline bluebird-api/bluebird_api/routers/core.py]: Can we rename the endpoint now that there aren't two separate ones?


```suggestion
async def environment(  # noqa: ANN201
```

File: bluebird-api/tests/api/test_routes.py
Comment (AlvaroSierra) [inline bluebird-api/tests/api/test_routes.py:1]: Similar to my previous comment, can we rename the test_complete_environment_* to test_environment_*.

File: bluebird-dt/bluebird_dt/simulator/simulator.py [hunk 1]
```diff
@@ -392,21 +389,21 @@ def environment(
...
 
-        # gather the basic env data
+        # gather the basic env data, remove wind field data to reduce the size of the data
         env_data = {
             "time": env.time,
             "start_time": env.start_time,
             "aircraft": {callsign: aircraft.data() for (callsign, aircraft) in env.aircraft.items()},
             "coordinations": coordinations,
-            "wind_field": env.wind_field,
+            "wind_field": None,
```
Comment (AlvaroSierra) [inline bluebird-dt/bluebird_dt/simulator/simulator.py]:
```suggestion
```

File: bluebird-api/bluebird_api/routers/core.py [hunk 2]
```diff
@@ -107,30 +107,24 @@ async def complete_environment(  # noqa: ANN201
...
+@core_router.get("/wind_field", tags=["State"])
+async def wind_field(runner: RunnerDep):  # noqa: ANN201
     """
-    Get the environment data for a given sector.
+    Get the wind field data for the environment.
     """
     if RunnerStore.current_runner is None or runner.sim is None:
         return {"exists": False}
     # HMI doesn't need to reload the environment again
     runner.sim.manager.reload_environment = False
```
Comment (AlvaroSierra) [inline bluebird-api/bluebird_api/routers/core.py]: This check is not required any more as the RunnerDep checks if it exists in the store.


```suggestion
```

File: bluebird-api/bluebird_api/routers/core.py [hunk 3]
```diff
@@ -107,30 +107,24 @@ async def complete_environment(  # noqa: ANN201
...
 
-@core_router.get("/environment/{sector_id}", tags=["State"])
-async def environment(  # noqa: ANN201
-    runner: RunnerDep, sector_id: str | None = None, no_airspace: bool = False, last_n_observations: int = 0
-):
+@core_router.get("/wind_field", tags=["State"])
+async def wind_field(runner: RunnerDep):  # noqa: ANN201
     """
-    Get the environment data for a given sector.
+    Get the wind field data for the environment.
```
Comment (AlvaroSierra) [inline bluebird-api/bluebird_api/routers/core.py]:
```suggestion
    Get the wind field data for current environment.
```

File: bluebird-dt/bluebird_dt/core/wind.py [hunk 4]
```diff
@@ -109,20 +108,35 @@ class WindField(BaseModel):
     # This line allows us to use numpy arrays as fields.
     model_config = ConfigDict(arbitrary_types_allowed=True)
 
-    u_comp: np.ndarray
-    v_comp: np.ndarray
+    u_comp: np.ndarray | list
+    v_comp: np.ndarray | list
```
Comment (AlvaroSierra) [inline bluebird-dt/bluebird_dt/core/wind.py]: Generic required, could it also be added for the ndarray.

In what cases would this be a list? This different will probably add quite

File: bluebird-dt/bluebird_dt/core/wind.py
Comment (AlvaroSierra) [inline bluebird-dt/bluebird_dt/core/wind.py:1]: Do you mind fixing some of the type annotations already available in this file.

File: bluebird-dt/bluebird_dt/core/wind.py [hunk 5]
```diff
@@ -109,20 +108,35 @@ class WindField(BaseModel):
...
+    wind_speed: np.ndarray | list[float] | float | None = None
+    wind_direction: np.ndarray | list[float] | float | None = None
+
+    # Validation and serialisation methods for numpy arrays, since these aren't natively supported by Pydantic
+    @field_validator("u_comp", "v_comp", "pressure_array", "lat_array", "lon_array")
+    def convert_to_array(cls, v: list | np.ndarray) -> np.typing.NDArray[np.float64]:
         return np.asarray(v, dtype=np.float64)
 
+    @field_validator("wind_speed", "wind_direction")
+    def convert_optional_to_array(cls, v: list | np.ndarray | float | None) -> np.typing.NDArray[np.float64] | None:
```
Comment (AlvaroSierra) [inline bluebird-dt/bluebird_dt/core/wind.py]: Generalise of the float type to allow 32 bit.

```suggestion
    def convert_optional_to_array(cls, v: list | np.ndarray | float | None) -> np.typing.NDArray[np.floating] | None:
```

File: bluebird-dt/bluebird_dt/core/wind.py [hunk 6]
```diff
@@ -109,20 +108,35 @@ class WindField(BaseModel):
...
     lon_array: np.ndarray | list[float]
     interpolation_method: typing.Literal["trilinear", "nearest", "fl_interpolation"] = "nearest"
-    # Instantiation method information. Will be None if using the default constructor
-    __construct_data: dict = PrivateAttr(default={"method": None, "args": None})
 
-    # Always use numpy arrays internally
-    @field_validator("pressure_array", "lat_array", "lon_array")
-    def convert_to_array(cls, v: list[float] | np.ndarray) -> np.typing.NDArray[np.float64]:
+    wind_speed: np.ndarray | list[float] | float | None = None
+    wind_direction: np.ndarray | list[float] | float | None = None
```
Comment (AlvaroSierra) [inline bluebird-dt/bluebird_dt/core/wind.py]: Can this be a list or array? We should try to minimise the types of variables and parse on input/output.

**Review by AlvaroSierra — CHANGES_REQUESTED** [review]
Thank you Bilal, some minor comments to clean up the code and type annotations of the Wind class.

**Review by georgedeath — CHANGES_REQUESTED** [review]
Codex review:

1. `bluebird-hmi/src/api/api.ts` still exposes the removed sector environment route as `/environment/${queryArg.sectorId}`. Since `bluebird-api/bluebird_api/routers/core.py` removes `/environment/{sector_id}`, consumers of `useEnvironmentQuery` will now get a 404. Even if the hook is not actively used today, the generated API surface is stale and should either be regenerated/updated or the compatibility route should stay.

2. `bluebird-dt/bluebird_dt/simulator/simulator.py` still returns `env.forecast_wind_field` under the `forecast` key. The PR sets `wind_field` to `None`, but replacing `manager.observe()` with `manager.environment` means forecast wind can now still be serialized through the main `/environment` response when present. That seems to undercut the stated intent of keeping wind data out of the environment endpoint for payload size/performance. Should `forecast` also be `None` or serialized through a separate endpoint/path?

3. The new API test for `wind_field()` currently allows `None`, so it does not prove the endpoint returns serialized wind data when wind exists. It would be good to add a test that seeds a runner/environment with a non-null `wind_field` and asserts the returned payload has the expected model fields.

**Comment by bilal-sattar-nats** [comment]
Thanks both. I've made those changes as requested!

File: bluebird-api/bluebird_api/routers/core.py [hunk 7]
```diff
@@ -107,30 +107,32 @@ async def complete_environment(  # noqa: ANN201
...
-        sector_id=sector_id,
-        no_airspace=no_airspace,
-        last_n_observations=last_n_observations,
-    )
+    wind_field = runner.sim.manager.environment.wind_field
+    return {"exists": True, "wind_field": wind_field.data() if wind_field is not None else None}
+
+
+@core_router.get("/forecast_wind_field", tags=["State"])
+async def forecast_wind_field(runner: RunnerDep):  # noqa: ANN201
```
Comment (AlvaroSierra) [inline bluebird-api/bluebird_api/routers/core.py]:
```suggestion
async def forecast_wind_field(runner: RunnerDep) -> WindField | None:  
```

File: bluebird-api/bluebird_api/routers/core.py [hunk 8]
```diff
@@ -107,30 +107,32 @@ async def complete_environment(  # noqa: ANN201
...
+    """
+    Get the forecast wind field data for current environment.
+    """
+
+    forecast_wind_field = runner.sim.manager.environment.forecast_wind_field
+    return {
+        "exists": True,
+        "forecast_wind_field": (forecast_wind_field.data() if forecast_wind_field is not None else None),
+    }
```
Comment (AlvaroSierra) [inline bluebird-api/bluebird_api/routers/core.py]:
```suggestion
    return runner.sim.manager.environment.forecast_wind_field
```

File: bluebird-api/bluebird_api/routers/core.py [hunk 9]
```diff
@@ -107,30 +107,32 @@ async def complete_environment(  # noqa: ANN201
...
 
-    return runner.sim.environment(
-        sim_time=runner.sim.manager.environment.time,
-        sector_id=sector_id,
-        no_airspace=no_airspace,
-        last_n_observations=last_n_observations,
-    )
+    wind_field = runner.sim.manager.environment.wind_field
+    return {"exists": True, "wind_field": wind_field.data() if wind_field is not None else None}
+
```
Comment (AlvaroSierra) [inline bluebird-api/bluebird_api/routers/core.py]:
```suggestion
    return runner.sim.manager.environment.wind_field
```

File: bluebird-api/bluebird_api/routers/core.py [hunk 10]
```diff
@@ -107,30 +107,32 @@ async def complete_environment(  # noqa: ANN201
...
         last_n_observations=last_n_observations,
     )
 
 
-@core_router.get("/environment/{sector_id}", tags=["State"])
-async def environment(  # noqa: ANN201
-    runner: RunnerDep, sector_id: str | None = None, no_airspace: bool = False, last_n_observations: int = 0
-):
+@core_router.get("/wind_field", tags=["State"])
+async def wind_field(runner: RunnerDep):  # noqa: ANN201
```
Comment (AlvaroSierra) [inline bluebird-api/bluebird_api/routers/core.py]:
```suggestion
async def wind_field(runner: RunnerDep) -> WindField | None: 
```

File: bluebird-api/bluebird_api/routers/core.py
Comment (AlvaroSierra) [inline bluebird-api/bluebird_api/routers/core.py]: Remove, this is more required any more.

File: bluebird-api/tests/api/test_routes.py [hunk 11]
```diff
@@ -313,87 +316,147 @@ class TestState:
...
+            ]
 
-        @pytest.mark.parametrize("sector_id", ["SPRINGFIELD"])
+            received = await routers.core.wind_field(runner)
+
+            assert isinstance(received, dict)
+            assert received["exists"] is True
+            assert "wind_field" in received
+            if received["wind_field"] is not None:
+                assert sorted(received["wind_field"].keys()) == sorted(expected_wind_keys)
```
Comment (AlvaroSierra) [inline bluebird-api/tests/api/test_routes.py]: Why not just try to deserialise the wind field? Will be more full proof and require less maintenance.

**Review by AlvaroSierra — CHANGES_REQUESTED** [review]

**Review by georgedeath — CHANGES_REQUESTED** [review]
1. `SectorPanel` calls `useEnvironmentQuery(undefined, ...)`, but the generated `environment` query dereferences `queryArg.noAirspace` / `queryArg.lastNObservations` while building params. I reproduced this through the generated RTK Query endpoint: initiating the query with `undefined` fails with `TypeError: Cannot read properties of undefined (reading 'noAirspace')`, while `{}` gets past query construction. Since this is the actual `SectorPanel` call shape, could we either pass `{}` there or make the generated query tolerate an omitted arg?

2. The normal `WindField.artificial(..., interpolation_method="fl_interpolation")` path works, including scalar speed/direction. However, because this PR now supports full-model deserialization, a deserialized/directly validated `WindField` can have `interpolation_method="fl_interpolation"` with missing `wind_speed` / `wind_direction`, and `get_wind_vector_using_plevels(...)` then returns NaN components rather than rejecting the invalid state. Could we add validation that both fields are present whenever `interpolation_method == "fl_interpolation"`?

**Review by georgedeath — APPROVED** [review]

**Review by AlvaroSierra — APPROVED** [review]
